src/main/java/mate/academy/AsyncRequestProcessor.java

```diff
@@ -1,16 +1,36 @@
 package mate.academy;
 
+import java.util.Map;
 import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ConcurrentHashMap;
 import java.util.concurrent.Executor;
+import java.util.concurrent.TimeUnit;
 
 public class AsyncRequestProcessor {
+    private static final long TIMEOUT = 1000;
+    private static final String USER_DETAILS = "Details for %s";
+
     private final Executor executor;
+    private final Map<String, UserData> cache = new ConcurrentHashMap<>();
 
     public AsyncRequestProcessor(Executor executor) {
         this.executor = executor;
     }
 
     public CompletableFuture<UserData> processRequest(String userId) {
-        return null;
+        return CompletableFuture.supplyAsync(() ->
+           cache.computeIfAbsent(
+                   userId, this::getUserData), executor
+        );
+    }
+
+    private UserData getUserData(String userId) {
+        try {
+            TimeUnit.MILLISECONDS.sleep(TIMEOUT);
+            return new UserData(userId, USER_DETAILS.formatted(userId));
+        } catch (InterruptedException e) {
+            Thread.currentThread().interrupt();
+            throw new RuntimeException(e);
+        }
```

Comment:
The 'getUserData' method handles InterruptedException by re-interrupting the thread, which is a good practice. However, you should also consider the consequences of throwing a RuntimeException. If this exception is not handled properly, it could lead to unresponsive or unpredictable behavior in your application. It might be better to handle this exception in a way that does not disrupt the entire process, such as returning a default UserData object or a specific error code.